lib/path/default.nix

```
@@ -217,6 +217,45 @@ in /* No rec! Add dependencies on this file at the top. */ {
               second argument: "${toString path2}" with root "${toString path2Deconstructed.root}"'';
         take (length path1Deconstructed.components) path2Deconstructed.components == path1Deconstructed.components;
 
+  /*
+  Split the filesystem root from a path.
+  The result is an attribute set with these attributes:
+
+  - `root`: The filesystem root of the path, meaning that this directory has no parent directory.
+  - `subpath`: The normalised subpath string that leads to the given path from the filesystem root, see `lib.path.subpath.normalise`
+
+  Laws:
+
+  - Appending the `root` and `subpath` gives the original path:
+
+        append (splitRoot p).root (splitRoot p).subpath == p
```

```suggestion
        p ==
          append
            (splitRoot p).root
            (splitRoot p).subpath
```

Aligning the repeated subexpression makes this more readable.